### DISCUSSION TITLE
Git, Github and Code for i interoperability

### DISCUSSION THREAD

**Discussion by NorskieRider** [codefori/vscode-ibmi#1710] [IBM i] [answered]
I'm exploring how to use git and github with Code for i.

How have others done this?

VSCode has integrations for git and github, but it appears to place the repository in a folder local to the workstation.

Code for i works from remote libraries or directories on the IBM i. 

It doesn't appear that VSCode manages git on a remote system, unless perhaps through a mapped drive?

**Comment by worksofliam** [comment, marked as answer]
Hi @NorskieRider! Great question.

I have written tons in our docs on how to use git, IBM i and VS Code together.

Start here, and see all the pages under 'Local Development': https://codefori.github.io/docs/#/pages/developing/local/getting-started

**Reply by chrjorgensen** [reply]
Just a remark on your mentioning of GitHub:

GitHub is just a service for storing git repositories remote, plus it adds features like pull requests, issue tracking, discussions etc.
There are several of these services available - read more here: [https://www.softwaretestinghelp.com/github-alternatives/](https://www.softwaretestinghelp.com/github-alternatives/)

In the link above is mentioned `GitBucket`, which runs on Java - and you can install it locally on your IBM i:
[https://ibmi-oss-docs.readthedocs.io/en/latest/java/wildfly/DEPLOYMENT.html?highlight=gitbucket](https://ibmi-oss-docs.readthedocs.io/en/latest/java/wildfly/DEPLOYMENT.html?highlight=gitbucket)
[https://ibmi-oss-docs.readthedocs.io/en/latest/java/tomcat/DEPLOYMENT.html?highlight=gitbucket](https://ibmi-oss-docs.readthedocs.io/en/latest/java/tomcat/DEPLOYMENT.html?highlight=gitbucket)

IBM i shops has traditionally stored the source locally on the system and not outside it - using git and git server on IBM i makes it possible to continue doing this.

**Reply by NorskieRider** [reply]
This is excellent! Thank you!
